firmware/tools/run_formatter.sh

```diff
@@ -1,18 +1,78 @@
-#! /bin/sh
-
-IMAGE_NAME="buildenv"
+#! /bin/bash
 
 GIT_ROOT=$(git rev-parse --show-toplevel)
 FIRMWARE_ROOT=$GIT_ROOT/firmware
 BUILD_DOCKER_SCRIPT=$FIRMWARE_ROOT/tools/build_docker_image.sh
 
 TEMP_FIND_FILE=/workspace/formatter_files.txt
 
-source "$BUILD_DOCKER_SCRIPT"
+echoerr()
+{
+    echo "$@" 1>&2;
+}
+
+printHelp ()
+{
+    echo "###################### Commands ######################"
+    echo "### <NONE> -> run formatter and replace code with formatted output"
+    echo "### -c, --check -> Check if code is formatted correctly"
+    echo "### -h, --help -> Print help message"
+
+    exit 1
+}
+
+startDocker ()
+{
+    local IMAGE_NAME="buildenv"
+
+    source "$BUILD_DOCKER_SCRIPT"
+
+    docker create -it -v $FIRMWARE_ROOT:/workspace --name env $IMAGE_NAME bash > /dev/null
+    docker start env > /dev/null
+}
+
+cleanupDocker ()
+{
+    docker rm -f env > /dev/null
+}
+
+runFormatter ()
+{
+    # get a list of all the files we want to run the formatter on
+    docker exec -i env bash -c "find /workspace/. \( -name '*.c' -o -name '*.h' -o -path '/workspace/./dependencies' -prune \) -type f > $TEMP_FIND_FILE"
+
+    if [ $CHECK_FLAG == "TRUE" ]; then
```

Comment:
I think this should be double brackets

Reply:
single brackets here are fine, but I'm fine with making double brackets our standard if that's what we want. It's less portable apparently but I don't think that's a huge issue here

Reply:
There's an exception on this line when I run in mingw without double brackets

Reply:
oh really? that's odd. I'll change it

Reply:
I had that issue before with mingw, I'm not sure why it's like that